src/runtime/HalideBuffer.h

```diff
@@ -1951,55 +1951,42 @@ class Buffer {
     // @{
     template<int N>
     struct for_each_value_task_dim {
-        int extent;
-        int stride[N];
+        std::ptrdiff_t extent;
+        std::ptrdiff_t stride[N];
     };
 
     // Given an array of strides, and a bunch of pointers to pointers
     // (all of different types), advance the pointers using the
     // strides.
     template<typename Ptr, typename... Ptrs>
-    HALIDE_ALWAYS_INLINE static void advance_ptrs(const int *stride, Ptr *ptr, Ptrs... ptrs) {
+    HALIDE_ALWAYS_INLINE static void advance_ptrs(const std::ptrdiff_t *stride, Ptr *ptr, Ptrs... ptrs) {
```

Comment:
Idle thought: would using references rather than pointers here enable potentially better codegen?

Reply:
I doubt it matters, this is getting inlined. I actually would stylistically prefer references but I didn't want to change it in this PR.